ckanext/unhcr/actions.py

```diff
@@ -1123,15 +1097,163 @@ def access_request_list_for_user(context, data_dict):
+    :rtype: list of dictionaries
+    """
+    m = context.get('model', model)
+    user_id = toolkit.get_or_bust(context, "user")
+    status = data_dict.get("status", "requested")
+    if status not in ['requested', 'approved', 'rejected', 'all']:
+        raise toolkit.ValidationError('Invalid status {}'.format(status))
+
+    user = m.User.get(user_id)
+    if not user:
+        raise toolkit.ObjectNotFound("User not found")
+
+    toolkit.check_access('access_request_list_for_user', context, data_dict)
+
+    access_requests_table = m.meta.metadata.tables["access_requests"]
+    group_table = m.meta.metadata.tables["group"]
+    package_table = m.meta.metadata.tables["package"]
+    user_table = m.meta.metadata.tables["user"]
+
+    sql = _get_base_query_for_access_requsts(
+        access_requests_table,
+        group_table,
+        package_table,
+        user_table,
+    )
+
+    if status != 'all':
+        sql = sql.where(access_requests_table.c.status == status)
+
+    if user.sysadmin:
+        # Sysadmin can see all access requests
+        access_requests = [
+            dictize_access_request(req) for req in m.Session.execute(sql).fetchall()
+        ]
+    else:
+        # Non-sysadmin can only see access requests for objects they have admin access to
+        organizations = toolkit.get_action("organization_list_for_user")(
+            context, {"id": user_id, "permission": "admin"}
+        )
+
+        containers = [o["id"] for o in organizations]
+        if not containers:
+            access_requests = []
+        else:
+            sql = _update_access_requests_query_for_containers(
+                sql,
+                user,
+                access_requests_table,
+                package_table,
+                containers,
+            )
+
+            access_requests = [
+                dictize_access_request(req) for req in m.Session.execute(sql).fetchall()
+            ]
+
+    # Improve dicts with user names
+    user_names = _get_required_user_names(access_requests)
+    for request in access_requests:
+        if request['actioned_by']:
+            request['actioned_by_name'] = user_names[request['actioned_by']]
+        if request['object_type'] == 'user':
+            request['request_user_name'] = user_names[request['object_id']]
+
+    # Add pending requests for new containers (we don't use access_requests for those)
+    # This is only for sysadmins
+    if user.sysadmin:
+        _get_container_requests(context, user_names)
+
+        # Order by timestamp (avoid new containers request at the bottom)
+        access_requests = sorted(
+            access_requests, key=lambda k: k['timestamp'], reverse=True
+        )
 
-    return [dictize_access_request(req) for req in m.Session.execute(sql).fetchall()]
+    # add a representative string for each
+    for ar in access_requests:
+        if ar['object_type'] == 'user':
+            ar['object_title'] = ar['user']['fullname'] or ar['user']['fullname']
```

Comment:
Maybe you meant this?
```suggestion
            ar['object_title'] = ar['user']['fullname'] or ar['user']['name']
```